### ISSUE TITLE
Loss for training ROBERTA

### ISSUE THREAD

**Issue by gaopengcuhk** [pytorch/fairseq#1102]
<img width="1465" alt="Capture" src="https://user-images.githubusercontent.com/51525797/64138272-54204680-cdca-11e9-8bb3-e3e2ded642cb.PNG">

The photo is only one example. I notice the same effect on training on WIKI+BOOKS and wikitext.

When you finish one epoch and start another epoch, the loss will drop a lot. For example, loss at the end of epoch one is 15.191 while the loss in the beginning of epoch two is 13.721.

**Comment by myleott** [comment]
The reported loss is a running average for the epoch. So you often see a big drop at epoch boundaries when the average resets. We have future plans to change this behavior to average over the log-interval, but for now it's epoch-based.
